web-api/migration-terraform/main/lambdas/migrations/0027-contact-primary-secondary-service-preference.js

```diff
@@ -0,0 +1,150 @@
+          ExpressionAttributeValues: {
+            ':pk': item.pk,
+            ':prefix': 'privatePractitioner',
+          },
+          KeyConditionExpression: '#pk = :pk and begins_with(#sk, :prefix)',
+          TableName: process.env.SOURCE_TABLE,
+        })
+        .promise()
+        .then(res => {
+          return res.Items;
+        });
+
+      const caseRecord = aggregateCaseItems([item, ...privatePractitioners]);
+      let itemToModify = cloneDeep(item);
+
+      const currentCaseEntity = new Case(caseRecord, { applicationContext });
+
+      let shouldUpdateServiceIndicators = false;
+
+      // we check the service indicator again to prevent overwriting an intentional
+      // service preference since the outer conditional is evaluation BOTH contacts
+      const shouldUpdateContactPrimary =
+        contactHasServiceIndicatorNone(item.contactPrimary) &&
+        !currentCaseEntity.isUserIdRepresentedByPrivatePractitioner(
+          item.contactPrimary.contactId,
+        );
+      if (shouldUpdateContactPrimary) {
+        delete itemToModify.contactPrimary.serviceIndicator;
+        shouldUpdateServiceIndicators = true;
+      }
+
+      // we check the service indicator again to prevent overwriting an intentional
+      // service preference since the outer conditional is evaluation BOTH contacts
+      const shouldUpdateContactSecondary =
+        contactHasServiceIndicatorNone(item.contactSecondary) &&
+        !!(
+          item.contactSecondary &&
+          !currentCaseEntity.isUserIdRepresentedByPrivatePractitioner(
+            item.contactSecondary.contactId,
+          )
+        );
+
+      if (shouldUpdateContactSecondary) {
+        delete itemToModify.contactSecondary.serviceIndicator;
+        shouldUpdateServiceIndicators = true;
+      }
+
+      if (shouldUpdateServiceIndicators) {
+        // we only care about mutations to the contacts service preference
+        const {
+          contactPrimary,
+          contactSecondary,
+        } = setServiceIndicatorsForCase({
+          ...itemToModify,
+          privatePractitioners: [...caseRecord.privatePractitioners],
+        });
+
+        if (shouldUpdateContactPrimary && contactPrimary) {
+          // trying to reduce blast radius on what actually gets mutated on the item
+          itemToModify.contactPrimary = {
+            ...item.contactPrimary,
+            serviceIndicator: contactPrimary.serviceIndicator,
+          };
+        }
+
+        if (shouldUpdateContactSecondary && contactSecondary) {
+          // trying to reduce blast radius on what actually gets mutated on the item
+          itemToModify.contactSecondary = {
+            ...item.contactSecondary,
+            serviceIndicator: contactSecondary.serviceIndicator,
+          };
+        }
+
+        new Case(
+          { ...caseRecord, itemToModify },
+          {
+            applicationContext,
+          },
+        ).validate();
+
+        applicationContext.logger.info(
```

Comment:
@mmarcotte Just wanted to note these are the items to look for in the logs. I've added flags on whether contactPrimary or contactSecondary was updated for a little more info. Let me know if anything else would be helpful here.